image-rs/src/signature/mod.rs

```diff
@@ -4,44 +4,132 @@
 //
 
 pub mod image;
-pub mod mechanism;
 pub mod payload;
 pub mod policy;
 
-use crate::{config::Paths, signature::policy::Policy};
+use std::{path::Path, sync::Arc};
 
-use anyhow::Result;
+use image::Image;
+use policy::policy_requirement::PolicyReqType;
+pub use policy::Policy;
+
+use anyhow::{bail, Context, Result};
 use oci_client::secrets::RegistryAuth;
 
-/// `allows_image` will check all the `PolicyRequirements` suitable for
-/// the given image. The `PolicyRequirements` is defined in
-/// [`policy_path`] and may include signature verification.
-#[cfg(feature = "signature")]
-pub async fn allows_image(
-    image_reference: &str,
-    image_digest: &str,
-    auth: &RegistryAuth,
-    file_paths: &Paths,
-) -> Result<()> {
-    use crate::{resource, signature::image::Image};
-
-    let reference = oci_client::Reference::try_from(image_reference)?;
-    let mut image = Image::default_with_reference(reference);
-    image.set_manifest_digest(image_digest)?;
-
-    // Read the set of signature schemes that need to be verified
-    // of the image from the policy configuration.
-    let policy_json_string = resource::get_resource(&file_paths.policy_path).await?;
-    let mut policy = serde_json::from_slice::<Policy>(&policy_json_string)?;
-    let schemes = policy.signature_schemes(&image);
-
-    // Get the necessary resources from KBS if needed.
-    for scheme in schemes {
-        scheme.init(file_paths).await?;
+use crate::resource::ResourceProvider;
+
+/// Image security config dir contains important information such as
+/// security policy configuration file and signature verification configuration file.
+pub const IMAGE_SECURITY_CONFIG_SUBDIR: &str = "image-security";
+
+pub struct ImageSecurity {
```

Comment:
This is kind of an abstract name for a struct. Maybe it could be something that reflects the config like `ImageSecurityConfig` or something that reflects what it is doing like `ImageSecurityChecker` It seems like this mainly for signatures so maybe `SignatureValidator` or something could make sense.

Reply:
`ImageSignatureValidator` sounds good.